src/berry_mill/kiwrap.py

```diff
@@ -107,9 +120,17 @@ def _get_repokeys(self, reponame: str, repodata: Dict[str, str]) -> str:
         else:
             s_url = urljoin(f"{url.scheme}://{url.netloc}/{url.path}/Release.key", "")
             response = requests.get(s_url, allow_redirects=True)
-            with open(g_path, 'xb') as f_rel:
-                f_rel.write(response.content)
+            # check reponse OK
+            if response.status_code == HTTPStatus.OK:
+                with open(g_path, 'xb') as f_rel:
+                    f_rel.write(response.content)
+            else:
+                log.warning(f"Wrong url defined for repo {reponame}")
```

Comment:
`if response.status_code is not HTTPStatus.OK:`
 
you still return a g_path that will point to a file that does not even exist. 

Please make sure that only a valid path to a valid GPG is returned and otherwise an empty string (or Null, if you make the calling functions except a null return value of this function.)

Reply:
Well ok not quite... I think what happens is that a non exsting path is passed to _verifiy_gpg_check() which will throw an unnecessary error that the given file does not exist

Reply:
Please just return ""  immediately when the HTTPStatus is not OK

Reply:
done